File: C++/countingSort.cpp
//  A program implementing the counting sort algorithm in C++.

#include<iostream>
using namespace std;

int main() {
    int size, max=0;
    cout << "Enter the number of digits to be sorted: ";
    cin >> size;                                                //  Taking input for number of elements to be sorted.
    int input[size], output[size];
    cout << "Enter the digits: ";                               //  Taking input for actual elements.
    for(int i=0 ; i<size ; i++) {
        cin >> input[i];
        if (input[i] > max)
            max = input[i];                                     //  Simultaneously calculating the maximum value.
    }

    int frequency[max];                                         //  Initializing an array to holds frequencies.
    for(int i=0 ; i<max ; i++)
        frequency[i] = 0;                                       //  Setting the frequencies to zero.
    for(int i=0 ; i<size ; i++)
        frequency[input[i]-1]++;                                //  Incrementing frequency of the respective element.
    for(int i=1 ; i<max ; i++)
        frequency[i] = frequency[i] + frequency[i-1];           //  Calculating cumulative frequencies.
    for(int i=size-1 ; i>=0 ; i--) {
        output[frequency[input[i] - 1] - 1] = input[i];         //  Generating the output array by considering the cumulative frequencies. 
        frequency[input[i] - 1]--;
    }

    cout << "Sorted sequence: ";
    for(int i=0 ; i<size ; i++)                                 //  Printing out the sorted sequence of elements.
        cout << output[i] << ' ';
    
    return 0;
}

File: Python/CountingSort.py
if __name__=="__main__":

    #  Get number of elements 
    size=int(input('Enter the number of digits to be sorted:'))
     
    #  get elements in a list by iteration
    ls=[]
    for i in range(0,size ): 
        e = int(input()) 
        ls.append(e)  # add the element to list

    element_range = max(ls) - min(ls) + 1

    #  Setting all elements to zero
    freq = [0 for _ in range(element_range)] 
    listsorted = [0 for _ in enumerate(ls)] 
  
    #  Count storing
    for i,j in enumerate(ls): 
        freq[ls[i]-min(ls)] =freq[ls[i]-min(ls)]+ 1
  
    #  Position elements 
    for i in range(1, len(freq)): 
        freq[i] += freq[i-1] 
  
    #  Setting to output array 
    for i in range(len(ls)-1, -1, -1): 
        listsorted[freq[ls[i] - min(ls)] - 1] = ls[i] 
        freq[ls[i] - min(ls)] -= 1
  
    print("Sorted array is " + str(listsorted)) 
     
